### DISCUSSION TITLE
Feed Direction

### DISCUSSION THREAD

**Discussion by mikedilger** [mikedilger/gossip#484] [Ideas]
Historically, before web browsers, chat clients worked the same as terminals:  You enter new elements at the bottom, and posts scroll upwards and off the screen.  This is also true of SMS clients on phones, of IRC, of AOL IM, of trillian, of discord, etc.

When Twitter came around, due to HTML peculiarities, they made an infinite-scroll system.  In that system, in order to start off at the newest stuff, the newest stuff had to be on top.  So the posting box was also at the top.  It was upside down, and everybody felt like it was upside down.  But they had to do it.  And people got used to it.

Now we are all confused because of what Twitter did.  But in most systems, the newest post is at the bottom, and history scrolls off the top. But also, when you start up the client, you don't see the history (except in discord).  In gossip, when you start the client you will already have a long history to explore.

I think we should flip the feed over, not just for DMs, but for all the feeds.  I think it's strange to have some feeds go one way and some go the other.  It's even stranger to have the newest posts on top, but the draft box on the bottom... the draft box should be where the newest post is written to.

**Comment by dtonon** [comment]
The "best" UI/UX, in the definition of "users find it pleasant and productive", has some primitive cognitive influences, but it is marked above all by habits.
In the past 25 years, when browsers take the scene, general contents are ordered from top to bottom. It is not only about Twitter, every news, forum and blog website uses this paradigm. Instead, messaging kept the opposite flow.
Forums are a nice all-in-one example, because the main list has recent discussions at the top, while the inner thread puts the last post on the bottom. 

I think these different patterns are related to the *connection* between the subsequent pieces of content: in the first case they are usually disconnected, so the interface promotes a view where the freshest items have more visibility, going back to explore history is an option, not a strict need.
In the latter, the UI asks the user to follow the classic (western) top-down reading path, to follow the conversation, to have better context and understanding.

I would prefer keeping the feed like it is now. Instead, we should reverse the DM chats.
You are right, it is strange to have some feeds go one way and some go the other, this probably pop up because the views have the same size. Usually, messaging interfaces are smaller, and the user can immediately spot the difference and get a sense of the different way to interact.
This is a area where we can apply the two-column layout productively: it will improve usability (one click fast switch) and reducing the feed width it will offer a distinctive layout that is easy to recognise.
